partone.asm

```diff
@@ -17,21 +17,34 @@
 	.data
 
 fibs:	.word   0 : 9         # create an array variable named "fibs" of 9 word-length elements (4 bytes each)
-size:	.word  9              # create a single integer variable named "size" that indicates the length of the array
+#	size:	.word             # create a single integer variable named "size" that indicates the length of the array
+prompt: .asciiz "Enter the size of the array to print: "
 
 ###############################################
 # .text segment. Assembly instructions go here.
 ###############################################
 	.text
-	      la   $s0, fibs        # load address of array into $s0
-	      la   $s5, size        # load address of size variable into $s5
-	      lw   $s5, 0($s5)      # load array size from its address in the register
+	      la	$s0, fibs        	# load address of array into $s0
+	      
+input:      li 	$v0, 4		# prep the system to print string
+		la	$a0, prompt		# load the input prompt into memory
+	      syscall			# print the prompt
+	      
+	      li	$v0, 5		# prep the system to read variable
+	      syscall
+	      
+	      add	$s5, $s5, $v0	# move the entered variable into a register
```

Comment:
-5 Because your input loop keeps adding $s5 to itself, if the user enters a negative number, then a valid number, line 36 add them together and you get less than you wanted to print.